engine/src/ast/function_expr.rs

```diff
@@ -10,7 +10,7 @@ use types::{GetType, LhsValue, RhsValue};
 #[serde(tag = "kind", content = "value")]
 pub(crate) enum FunctionCallArgExpr<'s> {
     LhsFieldExpr(LhsFieldExpr<'s>),
-    Literal(RhsValue),
+    Literal(LhsValue<'s>),
```

Comment:
As discussed privately, let's change this one back.

Reply:
I re-introduced it in a following commit (i'll squash them together before merge), but i don't really understand why its better than just keeping LhsValue as done in the previous commit?

Reply:
As seen from the JSON, `LhsValue` doesn't have syntactic information and misrepresents strings as bytes, while this distinction can be important both for external JSON consumers as well as for some function implementations that want to accept only strings and not raw bytes.